### ISSUE TITLE
Mac OS Sequoia permissions issues

### ISSUE THREAD

**Issue by dmurvihill** [guidepup/guidepup#137]
When running the Playwright example, Guidepup fails to start VoiceOver.

OS X 15.7.3 (24G419)
Guidepup 0.33
@guidepup/playwright 0.19

```
> papertrace-web@1.0.16-dev test:e2e
> playwright install --with-deps && playwright test guidepup.spec


Running 1 test using 1 worker

  ✘  1 …laywright VoiceOver › I can navigate the Guidepup Github page (2.6s)


  1) [webkit] › tests/e2e/guidepup.spec.ts:5:3 › Playwright VoiceOver › I can navigate the Guidepup Github page 

    Error: VoiceOver cannot be started
        at VoiceOver.start (/Users/dolan/IdeaProjects/papertrace/code/node_modules/@guidepup/guidepup/lib/macOS/VoiceOver/VoiceOver.js:295:19)
        at Object.voiceOver (/Users/dolan/IdeaProjects/papertrace/code/node_modules/@guidepup/playwright/lib/voiceOverTest.js:95:39)
    [cause]: Error: Failed to mount Guidepup preferences

    Please ensure you have run:

        - `npx @guidepup/setup setup` at least once on your machine to configure the OS
        - `npx @guidepup/setup install` at least once for this project to install screen reader assets
        at createPortableSymlinks (/Users/dolan/IdeaProjects/papertrace/code/node_modules/@guidepup/guidepup/lib/macOS/VoiceOver/preferences/createPortableSymlinks.js:27:19)
        at mountGuidepupPreferences (/Users/dolan/IdeaProjects/papertrace/code/node_modules/@guidepup/guidepup/lib/macOS/VoiceOver/preferences/mountGuidepupPreferences.js:22:57)
        at VoiceOver.start (/Users/dolan/IdeaProjects/papertrace/code/node_modules/@guidepup/guidepup/lib/macOS/VoiceOver/VoiceOver.js:274:56)
        at Object.voiceOver (/Users/dolan/IdeaProjects/papertrace/code/node_modules/@guidepup/playwright/lib/voiceOverTest.js:95:39)
    [cause]: Error: EEXIST: file already exists, symlink '/Volumes/GuidepupVoiceOverPreferences/VoiceOver/VoiceOver4.portable' -> '/Users/dolan/Library/Group Containers/group.com.apple.VoiceOver/Library/Preferences/com.apple.VoiceOver4.portable.scrd'
        at createPortableSymlinks (/Users/dolan/IdeaProjects/papertrace/code/node_modules/@guidepup/guidepup/lib/macOS/VoiceOver/preferences/createPortableSymlinks.js:24:39)
        at mountGuidepupPreferences (/Users/dolan/IdeaProjects/papertrace/code/node_modules/@guidepup/guidepup/lib/macOS/VoiceOver/preferences/mountGuidepupPreferences.js:22:57)
        at VoiceOver.start (/Users/dolan/IdeaProjects/papertrace/code/node_modules/@guidepup/guidepup/lib/macOS/VoiceOver/VoiceOver.js:274:56)
        at Object.voiceOver (/Users/dolan/IdeaProjects/papertrace/code/node_modules/@guidepup/playwright/lib/voiceOverTest.js:95:39)

    Error Context: test-results/tests-e2e-guidepup-Playwri-c6809-te-the-Guidepup-Github-page-webkit/error-context.md

  1 failed
    [webkit] › tests/e2e/guidepup.spec.ts:5:3 › Playwright VoiceOver › I can navigate the Guidepup Github page                                          
npm error Lifecycle script `test:e2e` failed with error:
npm error code 1
npm error path /Users/dolan/IdeaProjects/papertrace/code/app
npm error workspace papertrace-web@1.0.16-dev
npm error location /Users/dolan/IdeaProjects/papertrace/code/app
npm error command failed
npm error command sh -c playwright install --with-deps && playwright test guidepup.spec
```

When running setup per instructions, I had to grant full-disk access to my Terminal, but still got an error:
```
dolan@Dolans-MacBook-Pro app % npx @guidepup/setup setup      
Error: stepping, attempt to write a readonly database (8)

Environment setup complete 🎉
```

And the same issue persists.

**Comment by cmorten** [comment]
Hey @dmurvihill 👋 

Thanks for the issue report.

I've commented on the other one you've raised at https://github.com/guidepup/setup/issues/61 - this is more a confusing warning than a true setup error, hopefully addressed there but let me know what think!

Here the issue appears to be a genuine bug in Guidepup settings mounting stage (triggered by a `start()`), see the error cause in your log:

```console
    [cause]: Error: EEXIST: file already exists, symlink '/Volumes/GuidepupVoiceOverPreferences/VoiceOver/VoiceOver4.portable' -> '/Users/dolan/Library/Group Containers/group.com.apple.VoiceOver/Library/Preferences/com.apple.VoiceOver4.portable.scrd'
        at createPortableSymlinks (/Users/dolan/IdeaProjects/papertrace/code/node_modules/@guidepup/guidepup/lib/macOS/VoiceOver/preferences/createPortableSymlinks.js:24:39)
        at mountGuidepupPreferences (/Users/dolan/IdeaProjects/papertrace/code/node_modules/@guidepup/guidepup/lib/macOS/VoiceOver/preferences/mountGuidepupPreferences.js:22:57)
        at VoiceOver.start (/Users/dolan/IdeaProjects/papertrace/code/node_modules/@guidepup/guidepup/lib/macOS/VoiceOver/VoiceOver.js:274:56)
        at Object.voiceOver (/Users/dolan/IdeaProjects/papertrace/code/node_modules/@guidepup/playwright/lib/voiceOverTest.js:95:39)
```

Pertinent file: https://github.com/guidepup/guidepup/blob/main/src/macOS/VoiceOver/preferences/createPortableSymlinks.ts

Some investigation needed why the cleanup isn't working!

**Comment by cmorten** [comment]
Because we swallow errors from removal it's difficult to know why the symlinks still exist at the next step...

https://github.com/guidepup/guidepup/blob/main/src/macOS/VoiceOver/preferences/createPortableSymlinks.ts#L20

Could be `EPERM` / `EACCES` related which would be less than ideal if turns out this path is generally protected (hasn't been as far as can tell...)

The alternative is perhaps if VoiceOver is already running at this point (failed teardown from previous run?) then it could be holding onto, or recreating these links causing the `EEXIST` when try to recreate them?

All speculation, investigation needed!

Some sane steps could be:

1. Consider if removal should throw, not swallow - if not, whether need to introduce DEBUG logging for this kind of stuff
2. Could check for EEXIST errors where the link is as desired and treat as no-op (in catch or defensive check before hand for existing and correct so no-op?)
3. Ensure `start()` first stops any existing VO instance before proceeding with the settings and startup steps
4. Look at whether any auto-teardown options where a Node process has ended without a `stop()` being called

**Comment by dmurvihill** [comment]
VoiceOver still tearing down or failed teardown sounds like a good theory diagnosis; this doesn't happen every time and anecdotally, it appears to occur most often when cancelling a test and then re-running.

EDIT: However, I did just run it for the first time all day, and got the EEXIST error.

**Comment by cmorten** [comment]
@dmurvihill fixes applied to `@guidepup/setup`, `@guidepup/guidepup` and `@guidepup/playwright`.

Please let me know if still encountering the above issue and can re-open.
